### DISCUSSION TITLE
Over fitment of pins

### DISCUSSION THREAD

**Discussion by Manoj-Kumar-MMU** [stnolting/neorv32#1074] [General]
I want to upload the core to my intel DE10-Lite FPGA Board but when I synthesized the top file it returns me 

`Error (176205): Can't place 531 pins with 2.5 V I/O standard because Fitter has only 350 such free pins available for general purpose I/O placement`

All other variable range are mostly in default values.

**Comment by Manoj-Kumar-MMU** [comment]
Summary of the compile report

![image](https://github.com/user-attachments/assets/df6935f4-c835-45a6-bb45-99a978837d0f)

**Comment by Manoj-Kumar-MMU** [comment]
I have reduced some of the port vectors for interface that I wont be using (as attached in picture below). Not sure if its the proper way to solve the problem but it works for me

![image](https://github.com/user-attachments/assets/b5114c3c-ee2d-4ccb-9c8e-b812152c8ae6)

**Comment by stnolting** [comment]
Hey @Manoj-Kumar-MMU!

You should not use the processor's top module as top module for your FPGA setup. Better create a custom one and instantiate the processor. Then, you can customize the processor there and just propagate the actually required signals to actual FPGA pins.

There are also some pre-defined FPGA top's that you could use for getting started: [`rtl/test_setups`](https://github.com/stnolting/neorv32/tree/main/rtl/test_setups)

**Reply by Manoj-Kumar-MMU** [reply]
Thank you, managed to include some of the peripherals signals I needed.
